### PR TITLE
Add Magnetic units and conversions

### PR DESCRIPTION
**Pull request by samjrholt**
### Description

This pull request is to address missing magnetic CGS unit and equivalency. We have added Oersted as a CGS unit and added equivalency from Tesla to A/m.

Fixes #<Issue Number>

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: docs/changes/units/15962.feature.rst
Comment (pllim) [inline docs/changes/units/15962.feature.rst:1]: Change log failed to mention equivalency that you also added.

**Comment by pllim** [comment]
Some failures might be from the newly released pytest 8.

**Review by mhvk — COMMENTED** [review]
Thanks! I think the addition of the Oersted is a very good idea.

I'm less sure about the equivalency, as that would seem to be one very rarely used in astronomy -- and one always has to weigh the ease any addition provides against the difficulty in finding it and how it makes finding other things less easy. (If we do go ahead with it, it needs to be documented in `docs/units/equivalencies.rst`.)

One possibility would be to just do the `Oe` here and discuss the equivalency in a separate PR.

**Comment by eerovaher** [comment]
Haven't we avoided implementing CGS electromagnetism units because there are multiple different conventions?

**Comment by mhvk** [comment]
> Haven't we avoided implementing CGS electromagnetism units because there are multiple different conventions?

Yes, indeed, but the `Oersted` seems to be convention independent, at least from my scan of [wikipedia](https://en.wikipedia.org/wiki/Oersted), like `Gauss` and `Maxwell`. But I may be wrong...

File: docs/changes/units/15962.feature.rst
Comment (samjrholt) [reply docs/changes/units/15962.feature.rst:1]: I have removed this equivalency for now and will open a new PR for this.

**Comment by samjrholt** [comment]
> Thanks! I think the addition of the Oersted is a very good idea.
> 
> I'm less sure about the equivalency, as that would seem to be one very rarely used in astronomy -- and one always has to weigh the ease any addition provides against the difficulty in finding it and how it makes finding other things less easy. (If we do go ahead with it, it needs to be documented in `docs/units/equivalencies.rst`.)
> 
> One possibility would be to just do the `Oe` here and discuss the equivalency in a separate PR.

I have removed the equivalency from this PR to keep a separation of tasks and will create a new PR on this matter.

**Comment by samjrholt** [comment]
> > Haven't we avoided implementing CGS electromagnetism units because there are multiple different conventions?
> 
> Yes, indeed, but the `Oersted` seems to be convention independent, at least from my scan of [wikipedia](https://en.wikipedia.org/wiki/Oersted), like `Gauss` and `Maxwell`. But I may be wrong...

I can confirm that `Oe` is a convention independent unit. You are right that there are some "units" in CGS electromagnetism that are in fact not units and are not particularly well defined (such as `emu`) but `Oe` does not fall into this category.

File: astropy/units/cgs.py [hunk 1]
```diff
@@ -162,6 +164,13 @@
     namespace=_ns,
     doc="Maxwell: CGS unit for magnetic flux",
 )
+def_unit(
+    ["Oe", "Oersted", "oersted"],
+    1e3 / (4 * np.pi) * si.A / si.m,
+    namespace=_ns,
+    prefixes=True,
```
[the remark this answers is not in the document]
Comment (samjrholt) [reply astropy/units/cgs.py:171]: kOe and MOe are relatively common.

**Comment by samjrholt** [comment]
@pllim How do I solve the `"whatsnew-needed" present but no What's New entry; please add one.` error given by CI?

**Comment by neutrinoceros** [comment]
@samjrholt you need to edit `docs/whatsnew/6.1.rst` with a description of your change :)

**Comment by pllim** [comment]
I think this still needs a test too? Feel free to consult the code on how existing units are tested. Thanks!

**Comment by mhvk** [comment]
Actually, I see I can set it as ready for review myself, so I'll set this to automerge - thanks again!

**Comment by pllim** [comment]
@mhvk , please use the "squash and merge" option (auto or not). Thanks, all!

**Comment by pllim** [comment]
Don't we need to add a test?

**Comment by mhvk** [comment]
> Don't we need to add a test?

Oops, yes, good point! Let's add a test, maybe one that converts from regular SI to Oe, just in case.

**Comment by samjrholt** [comment]
> > Don't we need to add a test?
> 
> Oops, yes, good point! Let's add a test, maybe one that converts from regular SI to Oe, just in case.

Where is the best place to add the test? I can't see a place in `units/tests` where CGS units are currently being tested.

**Comment by mhvk** [comment]
@samjrholt - actually, on second thought, it is not like we have explicit tests for all units, so let's just get this in. Thanks!

**Pull request merged**